src/Dialogs/ColorPreferencesDialog.vala

```diff
@@ -0,0 +1,403 @@
+/*
+* Copyright (c) 2020 elementary, Inc. (https://elementary.io)
+*
+* This program is free software; you can redistribute it and/or
+* modify it under the terms of the GNU Lesser General Public
+* License version 3, as published by the Free Software Foundation.
+*
+* This program is distributed in the hope that it will be useful,
+* but WITHOUT ANY WARRANTY; without even the implied warranty of
+* MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+* General Public License for more details.
+*
+* You should have received a copy of the GNU Lesser General Public
+* License along with this program; if not, write to the
+* Free Software Foundation, Inc., 51 Franklin Street, Fifth Floor,
+* Boston, MA 02110-1301 USA
+*/
+
+namespace Terminal.Dialogs {
+    public class ColorPreferences : Gtk.Dialog {
+
+        private Gtk.ListStore console_theme_store;
+        private Gtk.ComboBox console_theme_combobox;
+        private Gtk.Image contrast_image;
+
+        private Gtk.ColorButton color01_button; // Black
+        private Gtk.ColorButton color02_button; // Red
+        private Gtk.ColorButton color03_button; // Green
+        private Gtk.ColorButton color04_button; // Yellow
+        private Gtk.ColorButton color05_button; // Blue
+        private Gtk.ColorButton color06_button; // Magenta
+        private Gtk.ColorButton color07_button; // Cyan
+        private Gtk.ColorButton color08_button; // Light gray
+        private Gtk.ColorButton color09_button; // Dark gray
+        private Gtk.ColorButton color10_button; // Light Red
+        private Gtk.ColorButton color11_button; // Light Green
+        private Gtk.ColorButton color12_button; // Light Yellow
+        private Gtk.ColorButton color13_button; // Light Blue
+        private Gtk.ColorButton color14_button; // Light Magenta
+        private Gtk.ColorButton color15_button; // Light Cyan
+        private Gtk.ColorButton color16_button; // White
```

Comment:
This is matching the naming in Gogh. I think it makes sense since it is generally shorter and there are a number of themes where for example "blue" does not actually correspond to a blue color.